### PR TITLE
[doc] update document for JavaScript API reference

### PR DESCRIPTION
**Pull request by fs-eire**
**Description**:  This change uses typedoc to generate JavaScript API reference document.

related change: #9077

preview:
- API summary page: https://fs-eire.github.io/onnxruntime/docs/api/
- JavaScript doc page: https://fs-eire.github.io/onnxruntime/docs/api/js-api.html
- API reference: https://fs-eire.github.io/onnxruntime/docs/api/js/index.html

### CODE REVIEW

File: docs/api/js/index.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+<a href="#onnx-runtime-javascript-api" id="onnx-runtime-javascript-api" style="color: inherit; text-decoration: none;">
+  <h1>ONNX Runtime JavaScript API</h1>
+</a>
+<p>ONNX Runtime JavaScript API is a unified API for all JavaScript usages. It&#39;s dependency of the following NPM packages:</p>
+<ul>
+<li>onnxruntime-node</li>
+<li>onnxruntime-web</li>
+<li>onnxruntime-react-native</li>
+</ul>
+<p>This package (onnxruntime-common) is not designed for using directly. Please consider to install one of the NPM packages above according to target platform.</p>
```
Comment (hanbitmyths) [inline docs/api/js/index.html:11]: "This package" <- it's not here, so we can directly use "onnxruntime-common is not designed ....".

Comment (fs-eire) [reply docs/api/js/index.html:11]: this section is directly generated from js/common/README.md. maybe I can just remove the README section, or write a summary for the API document

**Review by EmmaNingMS — APPROVED** [review]

File: docs/api/js/index.html (hunk 1 above)
Comment (fs-eire) [reply docs/api/js/index.html:11]: This is updated, but need to wait for #9077 get merged and update the re-generated docs

Comment (fs-eire) [reply docs/api/js/index.html]: updated to latest

**Pull request merged**